### PR TITLE
Integrate breadcrumb component.

### PR DESCRIPTION
**Pull request by Dhruvi16**
## Description

This PR integrates the breadcrumb component to the website.

## Technical details

Basic logic used here is -
1. Extracting slugs in an array
2. Putting those slugs in the breadcrumb component
3. forming the URL from those slugs for the links.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Dhruvi16** [comment]
I have one doubt here that links other than `/` are appending to the url so that naturally goes to 404 page which is obviously not desired.

File: webpack/sass/main.scss [hunk 1]
```diff
@@ -30,6 +33,19 @@ $project-card-theme: $blue;
...
+  border-top: 4px solid $color-forest-green;
+  background-color: $color-lighter-gray;
+
+  .breadcrumb {
+    @extend .padding-bigger;
+  
+    .link {
+      @extend .has-color-forest-green;
+    }
+  }
```
Comment (dhruvkb) [inline webpack/sass/main.scss:46]: You shouldn't have to edit styles here. If the component does not match the styles, make the change in Vocabulary itself.

File: webpack/sass/main.scss [hunk 2]
```diff
@@ -1,4 +1,7 @@
 @import "../node_modules/@creativecommons/vocabulary/css/vocabulary.css";
+@import "../node_modules/@creativecommons/vocabulary/scss/spacing.scss";
+@import "../node_modules/@creativecommons/vocabulary/scss/color.scss";
+@import "../node_modules/@creativecommons/vocabulary/scss/typography.scss";
 
 //Color Variables
 $pumpkin: #e15803;
```
Comment (dhruvkb) [inline webpack/sass/main.scss:7]: What's `$pumpkin`?

File: webpack/sass/main.scss [hunk 3]
```diff
@@ -1,4 +1,7 @@
 @import "../node_modules/@creativecommons/vocabulary/css/vocabulary.css";
+@import "../node_modules/@creativecommons/vocabulary/scss/spacing.scss";
+@import "../node_modules/@creativecommons/vocabulary/scss/color.scss";
+@import "../node_modules/@creativecommons/vocabulary/scss/typography.scss";
```
Comment (dhruvkb) [inline webpack/sass/main.scss:4]: This does not seem to be in use.

File: themes/vocabulary_theme/templates/layout.html [hunk 4]
```diff
@@ -127,17 +129,52 @@
...
+  {% if this._path !=  '/'%}
+  <div class="breadcrumb-container">
+  <nav class="container breadcrumb caption bold" aria-label="breadcrumbs">
+    <ul>
+      {% set crumbs = [] %}
+      {% set current = {'crumb': this} %}
+      <!-- Extracting the slugs of URL -->
+      {% for i in this._path.split("/") %}
+      {% if current.crumb is not none %}
+      {% if crumbs.insert(0, current.crumb._slug) %}{% endif %}
```
Comment (dhruvkb) [inline themes/vocabulary_theme/templates/layout.html:143]: There's no block inside the `if`.

File: themes/vocabulary_theme/templates/layout.html [hunk 5]
```diff
@@ -127,17 +129,52 @@
...
+      {% if current.update({"crumb": current.crumb.parent}) %}{% endif %}
```
Comment (dhruvkb) [inline themes/vocabulary_theme/templates/layout.html:144]: There's no block inside the `if`.

File: themes/vocabulary_theme/templates/layout.html [hunk 6]
```diff
@@ -127,17 +129,52 @@
...
+      {% endif %}
+      {% endfor %}
+      {% for crumb in crumbs %}
+      <!-- Active link -->
+      {% if this._slug == crumb %}
+      <li class="is-active"><a aria-current="page displayed">{{ crumb | title | replace('-', ' ') }}</a></li>
+      {% else %}
```
Comment (dhruvkb) [inline themes/vocabulary_theme/templates/layout.html:151]: Always indent inside `if` and `for` blocks.
```suggestion
      {% if this._slug == crumb %}
        <li class="is-active"><a aria-current="page displayed">{{ crumb | title | replace('-', ' ') }}</a></li>
      {% else %}
```

File: themes/vocabulary_theme/templates/layout.html [hunk 7]
```diff
@@ -127,17 +129,52 @@
...
+      {% if crumb != 'blog' %}
+      <!-- Forming the URL using extracted slugs -->
+      {% set i = loop.index %}
+      {% set ns = namespace (link = '') %}
+      {% for j in range(i) %}
+      {% set ns.link = ns.link + crumbs[j] + '/' %}
+      {% endfor %}
```
Comment (dhruvkb) [inline themes/vocabulary_theme/templates/layout.html:158]: Another example of indentation.
```suggestion
      {% for j in range(i) %}
        {% set ns.link = ns.link + crumbs[j] + '/' %}
      {% endfor %}
```

File: themes/vocabulary_theme/templates/layout.html [hunk 8]
```diff
@@ -127,17 +129,52 @@
...
-      <div class="col-lg-9 col-md-9 col-sm-12 content-wrap">
-        <div class="page py-3">
-          <h1 class="page-title pb-3 mb-4">{% block header %}{{ this.title }}{% endblock %}</h1>
-          {% block body %}{% endblock %}
-          <a id="back-to-top" href="#top" class="btn btn-dark btn-sm" role="button">Back to top</a>
-        </div>
-      </div>
-    </div>
+
+  <!-- breadcrumb -->
```
Comment (dhruvkb) [inline themes/vocabulary_theme/templates/layout.html:133]:
```suggestion
  <!-- Breadcrumb -->
```

File: webpack/sass/main.scss (hunk 1 above)
Comment (Dhruvi16) [reply webpack/sass/main.scss:46]: This is link color and padding. And I guess that will be different for every website.

File: webpack/sass/main.scss (hunk 2 above)
Comment (Dhruvi16) [reply webpack/sass/main.scss:7]: It is the old code.

File: webpack/sass/main.scss (hunk 3 above)
Comment (Dhruvi16) [reply webpack/sass/main.scss:4]: This is to keep consistency with other PRs.

Comment (Dhruvi16) [reply webpack/sass/main.scss:4]: And color tokens are used.

**Pull request merged**